handler.go

```diff
@@ -84,8 +82,17 @@ func (ca *Cagent) Run(outputFile *os.File, interrupt chan struct{}) {
 		select {
 		case <-interrupt:
 			return
-		case <-time.After(secToDuration(ca.Config.Interval)):
-			continue
+		default:
+			if retries > 0 {
+				log.Infof("Run: hub connection error %d/%d, retrying in %v s", retries, ca.Config.OnHTTP5xxRetries, ca.Config.OnHTTP5xxRetryInterval)
```

Comment:
This whole logic of retries seems to be complicated, and it seems it wouldn't work as expected. For example, if retries is already > 0 and an error that is != ErrHubTooManyRequests, != ErrHubServerError happened then it will wait for OnHTTP5xxRetryInterval interval, instead of ca.Config.Interval. But it's not expected behavior, if I understood it correctly.

Reply:
Retries can only reach != 0 on ErrHubServerError. I don't understand how the corner case you describe could trigger.

Reply:
If I understood correctly:
1. run -> ErrHubServerError happend -> retries became 1 -> it slept for OnHTTP5xxRetryInterval
2. rerun -> err happend != ErrHubTooManyRequests and != ErrHubServerError. By current logic it will sleep for OnHTTP5xxRetryInterval and then rerun.

Shouldn't it sleep for ca.Config.Interval and then rerun in this case?

Reply:
We can extend the code further to handle the error cases like you lay them out, but it would only make the code more complicated and you just raised a comment about `this whole logic of retries seems to be complicated`.

In both cases delay will happen before a retry.

Reply:
You're right. I'll rework the error check some more

Reply:
But behavior would be completely incorrect (both delay and output would be incorrect) in this scenario. It's a bug. It should be fixed. It can be rewritten in a new simple way, without making this complicated logic even more complicated. 

Please tell, what is the expected behavior of this code?

Reply:
@martinlindhe if you would like I can help you to fix it and make it more simple.

Reply:
Please consider to do something similar to this:
```
	retries := 0
	retryIn := secToDuration(ca.Config.Interval)

	for {
		err := ca.RunOnce(outputFile, ca.Config.OperationMode == OperationModeFull)
		if err != nil {
			if err == ErrHubTooManyRequests {
				// for error code 429, wait 10 seconds and try again
				retryIn = 10 * time.Second
			} else if err == ErrHubServerError {
				// for error codes 5xx, wait for configured amount of time and try again
				retryIn = time.Duration(ca.Config.OnHTTP5xxRetryInterval) * time.Second
				retries++
				// log error only when retries number exceeds a given threshold
				if retries > ca.Config.OnHTTP5xxRetries {
					log.Error("Run: hub connection error.")
					// set back to 0
					retries = 0
				} else {
					log.Infof("Run: hub connection error %d/%d, retrying in %v s", retries, ca.Config.OnHTTP5xxRetries, ca.Config.OnHTTP5xxRetryInterval)
				}
			} else {
				log.Error(err)
			}
		}

		select {
		case <-interrupt:
			return
		case <-time.After(retryIn):
			continue
		}
	}
```

Reply:
Updated PR with this suggestion